--- /usr/bin/prunerepo	2019-04-01 18:02:57.000000000 +0000
+++ /usr/bin/prunerepo	2019-11-12 09:20:27.292341122 +0000
@@ -40,6 +40,7 @@
     '--refresh',
     '--queryformat="%{location}"',
     '--quiet',
+    '--setopt=skip_if_unavailable=False',
 ]
 
 get_latest_packages_cmd = get_all_packages_cmd + [ '--latest-limit=1' ]
@@ -117,6 +118,9 @@
     log_info('Removing obsoleted packages...')
     was_deletion = False
     latest_rpms = get_rpms(get_latest_packages_cmd)
+    if not latest_rpms:
+        log_info("No RPMs available")
+        return was_deletion
     all_rpms = get_rpms(get_all_packages_cmd)
     to_remove_rpms = set(all_rpms) - set(latest_rpms)
     for rpm in to_remove_rpms:
