--- /usr/lib/python2.7/site-packages/koschei/backend/__init__.py~	2017-06-13 21:30:55.485685712 +0200
+++ /usr/lib/python2.7/site-packages/koschei/backend/__init__.py	2017-06-13 21:32:44.862493921 +0200
@@ -99,8 +99,8 @@
     build = Build(package_id=package.id, state=Build.RUNNING)
     name = package.name
     build_opts = {}
-    if package.arch_override:
-        override = package.arch_override
+    if True:
+        override = package.arch_override or '^'
         if override.startswith('^'):
             excludes = override[1:].split()
             build_arches = get_config('koji_config').get('build_arches')
